src/main/scala/com/github/mrpowers/spark/fast/tests/DataframeUtil.scala

```diff
@@ -0,0 +1,115 @@
+package com.github.mrpowers.spark.fast.tests
+
+import com.github.mrpowers.spark.fast.tests.ufansi.Color.{DarkGray, Green, Red}
+import com.github.mrpowers.spark.fast.tests.ufansi.Str
+import org.apache.commons.lang3.StringUtils
+import org.apache.spark.sql.Row
+
+object DataframeUtil {
+
+  def showDataframeDiff(
+      header: (String, String),
+      actual: Array[Row],
+      expected: Array[Row],
+      truncate: Int = 20
+  ): String = {
+
+    val sb = new StringBuilder
+
+    val a = actual.zipAll(expected, Row(), Row())
+    val diff = a.map { case (a, e) =>
+      if (equals(a, e)) {
```

Comment:
is this case when actual is completely equals to expected? It's probably not reachable given that we only call this when not equal.

Reply:
Changed vals names to indicate that we are comparing rows

Reply:
Ah right. I misread that part. Thanks for improving the name